services/api/src/models/definitions/audit-log.json

```diff
@@ -0,0 +1,40 @@
+{
+  "attributes": {
+    "requestPath":  {
+      "type": "String",
+      "required": true
+    },
+    "requestNamespace":  {
+      "type": "String",
+      "required": true
+    },
+    "requestNormalizedPath":  {
+      "type": "String",
+      "required": true
+    },
+    "requestMethod":  {
+      "type": "String",
+      "required": true
+    },
+    "action":  {
+      "type": "String",
+      "required": true
+    },
+    "type":  {
+      "type": "String",
+      "enum": ["create", "update", "delete", "read"]
```

Comment:
maybe it doesn't need to be an enum, just a freeform field called `action`

Reply:
fyi I do have a free form field called action. 
The `type` field was to todo make it easier to query on e.g. find me all deletions or all updates done to a given object

Reply:
@dominiek
 
I think it depends on the `Regenerate Invoice` what does. Does it update the invoice object or does it create a new invoice object. (then it would either be update or create)

Let's say we are starting a background job e.g. generate Invoice`s`, where there is no objectId. 

I guess we could have an `generic` or `action`. Or just reuse e.g. the `create`.
 
I am having a hard time seeing what value it brings. Like does having a generic / action make it easier to query?

Reply:
Thinking about it more ... this is suppose to be http based, we got the request method and url ... i dont think we need type at all.